server/controllers/reset.js

```diff
@@ -0,0 +1,68 @@
+const User = require("../models/User");
+const Token = require("../models/Token");
+const asyncHandler = require("express-async-handler");
+const crypto = require('crypto');
+const bcrypt = require('bcryptjs');
+const sendMail = require('../utils/sendMail');
+
+exports.forgotPassword = asyncHandler(async (req, res) => {
+  try {
+    const user = await User.findOne({
+      email: req.body.email
+    });
+
+    if (!user) {
+      res.status(404);
+      throw new Error("Email not found");
+    }
+
+    const token = await Token.findOne({
+      userId: user._id
+    });
+    if (token) await token.deleteOne();
+
+    const resetToken = crypto.randomBytes(32).toString("hex");
```

Comment:
Maybe generate a token using the user's id and verify it when the password has been changed? (This will avoid the need  for the token to be stored in a document)

Reply:
If I'm not saving the token to the database, how will I manage expired tokens? Will the user be able to change their password whenever they want?

Reply:
We already have jwt installed on the BE, you can use it to generate a token and set an expiry time.

Reply:
If I use jwt, wouldn't users be able to generate multiple reset links?

Reply:
Yes, users will be able to generate multiple reset links (you can set a short expiry time). However, it will exclude the need to store/retrieve/delete documents on each password reset.